### PR TITLE
Added deprecation warning for urllib3.contrib.securetransport

### PR DESCRIPTION
**Pull request by IvanLauLinTiong**
This PR is to address issue #2692

### CODE REVIEW

File: src/urllib3/contrib/securetransport.py [hunk 1]
```diff
@@ -91,6 +92,14 @@
     _temporary_keychain,
 )
 
+warnings.warn(
+    "'securetransport' module is deprecated and will be removed "
```
Comment (sethmlarson) [inline src/urllib3/contrib/securetransport.py:96]: Let's use the full name of the module: `urllib3.contrib.securetransport`.

File: src/urllib3/contrib/securetransport.py [hunk 2]
```diff
@@ -91,6 +92,14 @@
...
+    "in a future release of urllib3 2.x. Read more in doc: "
+    "https://urllib3.readthedocs.io/en/stable/reference/contrib/securetransport.html",
+    DeprecationWarning,
```
Comment (sethmlarson) [inline src/urllib3/contrib/securetransport.py:99]: Let's use the `category=DeprecationWarning` keyword argument here.

File: src/urllib3/contrib/securetransport.py [hunk 3]
```diff
@@ -91,6 +92,14 @@
     _temporary_keychain,
 )
 
+warnings.warn(
+    "'securetransport' module is deprecated and will be removed "
+    "in a future release of urllib3 2.x. Read more in doc: "
```
Comment (sethmlarson) [inline src/urllib3/contrib/securetransport.py:97]: "Read more in this issue: ..."

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Few comments:

- Let's point people to this issue: https://github.com/urllib3/urllib3/issues/2681 in both the documentation and the warning message. That way we can provide updates if necessary on the situation.

Also note I rejected your Open Collective expense as the work hadn't been completed yet, hold off on creating one until the work is complete. Thanks!

**Comment by IvanLauLinTiong** [comment]
Hi @sethmlarson ,  I have addressed all your comments in latest commit.

**Review by sethmlarson — APPROVED** [review]
LGTM. Thanks for this!

**Pull request merged**